tidy3d/components/lumped_element.py

```diff
@@ -272,28 +378,504 @@ def _sheet_conductance(self):
+            return RLCNetwork._to_series_network_transfer_function(R, L, C)
+        return RLCNetwork._to_parallel_network_transfer_function(R, L, C)
+
+    @staticmethod
+    def _to_series_network_transfer_function(
+        R: float, L: float, C: float
+    ) -> tuple[tuple[float, ...], tuple[float, ...]]:
+        """Converts the RLC series network to a rational expression representing the
+        admittance of the network in the Laplace domain.
+        """
+        if R and C and L:
+            # RLC series network
+            a = (0, C, 0)
+            b = (1, R * C, L * C)
+        elif R and C:
+            # RC series network
+            a = (0, C)
+            b = (1, R * C)
+        elif R and L:
+            # RL series network
+            a = (1, 0)
+            b = (R, L)
+        else:
+            # LC series network
+            a = (0, C, 0)
+            b = (1, 0, L * C)
+        return (a, b)
+
+    @staticmethod
+    def _to_parallel_network_transfer_function(
+        R: float, L: float, C: float
+    ) -> tuple[tuple[float, ...], tuple[float, ...]]:
+        """Converts the RLC parallel network to a rational expression representing the
+        admittance of the network in the Laplace domain.
+        """
+        if R and C and L:
+            # RLC parallel network
+            a = (R, L, R * L * C)
+            b = (0, R * L, 0)
+        elif R and C:
+            # RC parallel network
+            a = (1, R * C)
+            b = (R, 0)
+        elif R and L:
+            # RL parallel network
+            a = (R, L)
+            b = (0, R * L)
+        else:
+            # LC parallel network
+            a = (1, 0, L * C)
+            b = (0, L, 0)
+        return (a, b)
+
+    def _to_medium(self, scaling_factor: float) -> PoleResidue:
+        """Converts the :class:`RLCNetwork` model directly into a :class:`PoleResidue` model
+        with proper scaling depending on the lumped element's dimensions."""
+        R = self.resistance
+        C = self.capacitance
+        L = self.inductance
+
+        # eps_infinity is set to 1, simply to avoid validation errors.
+        # The final equivalent medium to be added to the simulation needs to be combined with the
+        # background medium, where this value of 1 will be ignored and the value of the background
+        # medium will be taken.
+        if self._number_network_elements == 1:
+            # Lumped element is simply a resistor, inductor, or capacitor.
+            if R:
+                # Technically zeroth order network
+                med = Medium(permittivity=1, conductivity=(scaling_factor / R))
+                return PoleResidue.from_medium(med)
+            if C:
+                med = Medium(permittivity=1 + scaling_factor * C / EPSILON_0, conductivity=0)
+                return PoleResidue.from_medium(med)
+            if L:
+                # TODO would be nice to be able to set the damping term exactly to 0
```

Comment:
why do we need some nonzero loss here?

Reply:
You might remember me asking questions a while ago about supporting Drude medium with zero loss, or equivalently supporting a PoleResidue medium with second order pole at omega=0. At the moment, it is not possible, so I have to add a bit of loss.